main.go

```diff
@@ -0,0 +1,215 @@
+)
+
+// init is called on startup
+func init() {
+	var err error
+	NY, err = time.LoadLocation("America/New_York")
+	if err != nil {
+		log.Fatalf("Cannot load NYC tz => {%s}", err)
+	}
+
+}
+
+// configure runs before startup
+func configure() {
+	PG_USER = getOrElse("PG_USER", "adicu")
+	PG_PASSWORD = getOrElse("PG_PASSWORD", "")
+	PG_DB = getOrElse("PG_DB", "")
+	PG_HOST = getOrElse("PG_HOST", "localhost")
+	PG_PORT = getOrElse("PG_PORT", "5432")
+	PG_SSL = getOrElse("PG_SSL", "disable")
+}
+
+// getDate parses a filepath to get a date from the filename given the regex
+// declared in `filenameRegex`.
+func getDate(s string) (time.Time, error) {
+	return time.ParseInLocation(
+		datetimeFormat,
+		datetimeRegex.FindString(path.Base(s)),
+		NY)
+}
+
+// getOrElse checks the specified environment variable, returns the value if found, otherwise
+// will return the default value provided. If there is no default then makes a fatal log.
+func getOrElse(key, standard string) string {
+	if val := os.Getenv(key); val != "" {
+		return val
+	} else if standard == "" {
+		log.Fatalf("The environment variable, %s, must be set", key)
+	}
+	return standard
+}
+
+// dbConnect yanks db configurations from the environment variables and returns a postgres
+// connection
+func dbConnect() *sql.DB {
+	db, err := sql.Open("postgres",
+		fmt.Sprintf("user=%s password=%s dbname=%s host=%s port=%s sslmode=%s",
+			PG_USER,
+			PG_PASSWORD,
+			PG_DB,
+			PG_HOST,
+			PG_PORT,
+			PG_SSL,
+		))
+	if err != nil {
+		log.Fatalf("Error connecting to Postgres => %s", err.Error())
+	}
+	log.Printf("PQ Database connection made to %s", PG_DB)
+	return db
+}
+
+// handleFile processes new files
+//
+// The file is read into memory, parsed then inserted to the database.
+func handleFile(filename string, db *sql.DB) {
+	log.Printf("Processing, %s", filename)
+	fileContents, err := ioutil.ReadFile(filename)
+	if err != nil {
+		log.Printf("Failed to read in file, %s => %s", filename, err.Error())
+		return
+	}
+
+	tm, err := getDate(filename)
+	if err != nil {
+		log.Printf("Failed to parse date from file, %s, ignored.", filename)
+		return
+	}
+
+	data, err := parseData(tm, fileContents)
+	if err != nil {
+		log.Printf("Failed to parse data from %s => %s", filename, err.Error())
+		return
+	}
+
+	if err = dataset(data).insert(db); err != nil {
+		log.Printf("Failed to insert data from, %s => %s", filename, err.Error())
+	}
+}
+
+// Update the materialized views listed in `materializedViews`
+func updateViews(db *sql.DB) {
+	txn, err := db.Begin()
+	if err != nil {
+		log.Printf("ERR: failed to start pq txn for materialized view updates => %s", err.Error())
+		return
+	}
+
+	for _, view := range materializedViews {
+		if _, err = txn.Exec(fmt.Sprintf("REFRESH MATERIALIZED VIEW %s", view)); err != nil {
+			log.Printf("Failed to update materialized view, %s => %s", view, err.Error())
+		}
+	}
+
+	if nil == txn.Commit() {
```

Comment:
why is the `nil` on the left side here? wouldn't it be more common to put it on the right side?